src/pages/SignIn/SignIn.jsx

```diff
@@ -1,29 +1,65 @@
-import React from 'react';
+import React , { useState } from 'react';
+import axios from 'axios';
 import './SignIn.css';
+import { Link, useNavigate, useLocation } from 'react-router-dom';
+import { useUserAuth } from '../../context/userAuth-context';
+import { useToastContext } from '../../context/toastContext';
+
 
 export default function SignIn() {
+  const navigate = useNavigate();
+   const [email , setEmail] = useState();
+   const [password , setPassword] = useState();
+   const { setIsLogIn } = useUserAuth();  
+   const location = useLocation();
+   const from = location.state?.from?.pathname || "/" ;
+   const  notify  = useToastContext();
+
+   const loginHandler = async(e) => {
+     e.preventDefault();
+    try {
+      const response = await axios.post(`/api/auth/login`, {
+        email , password
+      });
+      if(response.status === 200){
+        localStorage.setItem("token", response.data.encodedToken);
+        setIsLogIn(true);
+        notify('You Are Successfully LogIn!' , {type:'info'});
+        navigate(from , {replace:true});
+      } 
+    } catch (err) {
+        notify('The email you entered is not Registered. Please SignUp!' ,{ type:'error'});
+    }
+  }
+
+  const guestLoginHandler = () => {
+    setEmail('guest1234@gmail.com');
+    setPassword('guest1234')
+  }
+
   return (
     <>
     <div className="login-container">
         <h3 className="login-heading">Account Information</h3>
         <div className="login-card">
-            <h2>Sign In</h2>
-            <form className="logIn-form">
-                <label for="email" aria-required="true">E-mail address<span>*</span></label>
-                <input type="email" name="user-email" id="login-eamil-input"/>
+            <h2>LogIn</h2>
+            <form className="logIn-form" onSubmit={loginHandler}>
```

Comment:
I think you can remove onSubmit={loginHandler} in the form tag as you have given the onClick={loginHandler} button.